server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSPerformSPOptionsCheckUtility.java

```diff
@@ -0,0 +1,326 @@
+        checkHasMandatoryOptions(ctx.SECURITY(), ctx, "SECURITY");
+        checkHasMandatoryOptions(ctx.REBUILD(), ctx, "REBUILD");
+    }
+    private void checkShutdown(CICSParser.Cics_perform_shutdownContext ctx) {
+        checkHasMandatoryOptions(ctx.SHUTDOWN(), ctx, "SHUTDOWN");
+        checkHasMutuallyExclusiveOptions("SDTRAN or NOSDTRAN", ctx.SDTRAN(), ctx.NOSDTRAN());
+        checkHasMutuallyExclusiveOptions("IMMEDIATE or TAKEOVER", ctx.IMMEDIATE(), ctx.TAKEOVER());
+        if (!ctx.DUMP().isEmpty()) {
+            List<TerminalNode> filteredNodes = ctx.children.stream().filter(TerminalNode.class::isInstance).map(element -> (TerminalNode) element)
+                    .filter(node -> node.getSymbol().getType() == CICSLexer.DUMP).collect(Collectors.toList());
+            checkDumpDuplicates(filteredNodes, ErrorSeverity.WARNING);
+        }
+        if (!ctx.IMMEDIATE().isEmpty()) {
+            checkHasIllegalOptions(ctx.RESTART(), "RESTART");
+            checkHasIllegalOptions(ctx.XLT(), "XLT");
+            checkHasIllegalOptions(ctx.PLT(), "PLT");
+            checkHasIllegalOptions(ctx.PLTNAME(), "PLTNAME");
+        } else if (!ctx.TAKEOVER().isEmpty()) {
+            checkHasIllegalOptions(ctx.NORESTART(), "NORESTART");
+            checkHasIllegalOptions(ctx.RESTART(), "RESTART");
+            checkHasIllegalOptions(ctx.XLT(), "XLT");
+            checkHasIllegalOptions(ctx.PLT(), "PLT");
+            checkHasIllegalOptions(ctx.PLTNAME(), "PLTNAME");
+        } else {
+            checkHasMutuallyExclusiveOptions("PLT or PLTNAME", ctx.PLT(), ctx.PLTNAME());
+            checkHasIllegalOptions(ctx.NORESTART(), "NORESTART");
+        }
+    }
+    private void checkSsl(CICSParser.Cics_perform_sslContext ctx) {
+        checkHasMandatoryOptions(ctx.SSL(), ctx, "SSL");
+        checkHasMandatoryOptions(ctx.REBUILD(), ctx, "REBUILD");
+    }
+    private void checkstatistics(CICSParser.Cics_perform_statisticsContext ctx) {
+        checkHasMandatoryOptions(ctx.STATISTICS(), ctx, "STATISTICS");
+        checkHasMandatoryOptions(ctx.RECORD(), ctx, "RECORD");
+        checkPrerequisiteIsMet(ctx.ALL(), ctx.RESETNOW(), ctx, "RESETNOW without ALL");
+        checkHasMutuallyExclusiveOptions("JOURNALNAME or JOURNALNUM", ctx.JOURNALNAME(), ctx.JOURNALNUM());
+        checkHasMutuallyExclusiveOptions("TRANCLASS or TCLASS", ctx.TRANCLASS(), ctx.TCLASS());
+    }
+    private void checkDumpDuplicates(List<TerminalNode> rules, ErrorSeverity severity) {
+        Set<Integer> set = new HashSet<>();
+
+        rules.forEach(
+                 child -> {
+                    if (child.getSymbol().getType() == CICSLexer.DUMP && !set.add(child.getSymbol().getType())) {
+                            throwException(severity,
+                                    getLocality(child),
+                                    "Excessive options provided for: ",
+                                    child.getText());
+                    }
+                });
```

Comment:
They are all DUMP tokens.
```suggestion
        if (rules.size() <= 1) return;
        rules.subList(1, rules.size()).forEach(
                 child -> {
                        throwException(severity,
                                getLocality(child),
                                "Excessive options provided for: ",
                                "DUMP");
                });
```